crates/c-api/include/wasmtime/trap.h

```diff
@@ -13,6 +13,38 @@
 extern "C" {
 #endif
 
+/**
+ * \brief Code of instruction trap.Enum #wasmtime_trap_code_t
+ *
+ * See #wasmtime_trap_code_enum for possible values.
+ */
+typedef uint8_t wasmtime_trap_code_t;
+
+enum wasmtime_trap_code_enum {
+  /// The current stack space was exhausted.
+  #define WASMTIME_TRAP_CODE_STACK_OVERFLOW
+  /// An out-of-bounds memory access.
+  #define WASMTIME_TRAP_CODE_MEMORY_OUT_OF_BOUNDS
+  /// A wasm atomic operation was presented with a not-naturally-aligned linear-memory address.
+  #define WASMTIME_TRAP_CODE_HEAP_MISALIGNED
+  /// An out-of-bounds access to a table.
+  #define WASMTIME_TRAP_CODE_TABLE_OUT_OF_BOUNDS
+  /// Indirect call to a null table entry.
+  #define WASMTIME_TRAP_CODE_INDIRECT_CALL_TO_NULL
+  /// Signature mismatch on indirect call.
+  #define WASMTIME_TRAP_CODE_BAD_SIGNATURE
+  /// An integer arithmetic operation caused an overflow.
+  #define WASMTIME_TRAP_CODE_INTEGER_OVERFLOW
+  /// An integer division by zero.
+  #define WASMTIME_TRAP_CODE_INTEGER_DIVISION_BY_ZERO
+  /// Failed float-to-int conversion.
+  #define WASMTIME_TRAP_CODE_BAD_CONVERSION_TO_INTEGER
+  /// Code that was supposed to have been unreachable was reached.
+  #define WASMTIME_TRAP_CODE_UNREACHABLE_CODE_REACHED
+  /// Execution has potentially run too long and may be interrupted.
+  #define WASMTIME_TRAP_CODE_INTERRUPT
+};
```

Comment:
I think here it's ok to skip the `enum { .. }` wrapper, and I think the value of each `#define` constant will need to be listed.

Reply:
Oops, I had a bunch of `#define`s first, then decided to model this after the config enums like 

https://github.com/bytecodealliance/wasmtime/blob/73fd702bb7cabdff156eaf4ef8db5b92b098583e/crates/c-api/include/wasmtime/config.h#L56-L64

...and didn't follow through properly 😬 I've updated the code using a proper enum instead of defines. What do you think?

Reply:
Eh I could go either way, I don't actually know enough about C to know whether these are really that different, and they feel like the same to me. I'm happy to defer to whichever you feel is more appropriate here!

Reply:
I'm not a C expert either -- it felt to be a tad more consistent going with the enum definition. Updated ✔️